gem/views.py

```diff
@@ -377,3 +377,60 @@ def clean_comment(self):
 
 
 CommentDetailsForm.clean_comment = clean_comment
+
+
+class ReportCommentView(FormView):
+    template_name = 'comments/report_comment.html'
+    form_class = ReportCommentForm
+
+    def render_to_response(self, context, **response_kwargs):
+        comment = MoloComment.objects.get(pk=self.kwargs['comment_pk'])
+
+        for report in comment.gem_comment.all():
+            if self.request.user.id == report.user_id:
+                return HttpResponseRedirect(
+                    reverse('already_reported',
+                            args=(self.kwargs['comment_pk'],)
+                            )
+                )
```

Comment:
1. The property name `gem_comment` on `MoloComment` is a bit confusing. It sounds like it's referring to another type of comment but actually it's referencing the `GemReportComment` model. I would change the `related_name` on the `comment` property of `GemReportComment` to something like `gem_comment_reports` (I think the default, if you didn't set the `related_name`, would be `gemreportcomment_set`? - which would also be fine). Read on before you do this.

2. Consider changing the model name from `GemReportComment` to `GemCommentReport`. It reads better and will probably take care of the above if you let it use the default `related_name`. `gemcommentreport_set` is pretty descriptive when you're looking at a `MoloComment` instance. If you do this rename please also rename the related `GemReportCommentModelAdmin` and `GemReportCommentAdmin` classes for consistency and remember to update your migration.

3. You should be able to filter for the user instead of looping over every report and checking the user id.